### DISCUSSION TITLE
[Core Protocol] Why is latency from tx submit -> confirm is so long? Is there a way to improve that?

### DISCUSSION THREAD

**Discussion by upnodedev** [ethereum-optimism/developers#89] [Core Protocol]
### Issue Description

We have deployed an OP Stack with both 1s and 2s block time and have noticed that it is actually take about 4 - 8s from send tx to confirm noti popup. Tested on Rabby wallet which is possibly the fastest wallet.

See this video for demonstration

https://youtu.be/jQRQXuzvutc

You can see that it take about 7s to confirm and get the receipt.

We have tried with 1s block time and achieving similar latency. 

We would like to know why it take so long and is there a way to improve it?

### Additional Information

See this video for demonstration

https://youtu.be/jQRQXuzvutc

**Comment by sbvegan** [comment]
I'm not sure, two things come to mind:

- My initial assumption is that this is due to the wallet provider
- It might have to do with Goerli being deprecated. Do you see the same behavior on OP Sepolia and OP Mainnet?

Have you been able to bench mark this with a transaction directly submitted to the sequencer endpoint and see similar results? I'll see if someone else can weigh in on the matter as well.

**Reply by roninjin10** [reply]
This is correct that the RPC node being connected to are what are returning the receipt. I would suggest testing against non deprecated networks and also testing multiple RPC providers. Also consider testing other networks with similar fast block times.

**Reply by upnodedev** [reply]
We got about 5s on OP Mainnet from wallet confirmation -> tx confirmed popup

https://youtu.be/c-HvcXrjiDY

**Reply by sbvegan** [reply]
So the latency that we see on our public endpoints is generally between 2.5-3s. It seems like the confirmation is largely RPC dependent. You can try updating your wallets RPCs to something known to be faster. There is probably additional latency added in the wallet logic as well.

**Reply by sbvegan** [reply]
Hey @upnodedev just checking back in here too. Does this answer your question?

**Reply by upnodedev** [reply]
Hi @sbvegan can you show how you measure and get 2.5-3s

I have written some code that is similar to what we are implementing to measure and get these result

https://runkit.com/upnodedev/6567fbc7ae0041000845df4f

* OP Goerli: 6 - 12s
* OP Sepolia: 3 - 8s
* XAI (Arbitrum L3): 1.5 - 6s

**Reply by sbvegan** [reply]
We have an internal grafana dashboard that tracks the latency of all the requests that go to our public user facing endpoints. That dashboard uses the metrics that are enabled directly on the nodes that we're running. Here's an [overview](https://docs.optimism.io/builders/node-operators/metrics) of how we enable metrics on our rollup node.

**Reply by sbvegan** [reply]
Hey @upnodedev, I'm going to go ahead and close this thread. This RPC specific and not something on the core protocol.
